tools/ReleaseTools.psm1

```diff
@@ -6,40 +6,26 @@
 using module PowerShellForGitHub
 using namespace System.Management.Automation
 
-class RepoNames: IValidateSetValuesGenerator {
-    # NOTE: This is super over-engineered, but it was fun.
-    static [string[]] $Values = "vscode-powershell", "PowerShellEditorServices"
-    [String[]] GetValidValues() { return [RepoNames]::Values }
-}
-
-$ChangelogFile = "CHANGELOG.md"
+Import-Module $PSScriptRoot/VersionTools.psm1
 
 <#
 .SYNOPSIS
-  Given the repository name, execute the script in its directory.
+  Creates and checks out `release` if not already on it.
 #>
-function Use-Repository {
-    [CmdletBinding()]
+function Update-Branch {
+    [CmdletBinding(SupportsShouldProcess)]
     param(
         [Parameter(Mandatory)]
         [ValidateSet([RepoNames])]
-        [string]$RepositoryName,
-
-        [Parameter(Mandatory)]
-        [scriptblock]$Script
+        [string]$RepositoryName
     )
-    try {
-        switch ($RepositoryName) {
-            "vscode-powershell" {
-                Push-Location -Path "$PSScriptRoot/../"
-            }
-            "PowerShellEditorServices" {
-                Push-Location -Path "$PSScriptRoot/../../PowerShellEditorServices"
+    Use-Repository -RepositoryName $RepositoryName -Script {
+        $Branch = git branch --show-current
+        if ($Branch -ne "release") {
+            if ($PSCmdlet.ShouldProcess("release", "git checkout -B")) {
```

Comment:
❤️ love a good should process impl

Reply:
No idea if it actually still works, I never use it lmao.